src/services/session/SessionService.ts

```diff
@@ -13,6 +13,19 @@ export default class SessionService {
     req.session!.setExtraData(SUPPRESSION_DATA_KEY, updatedSession);
   }
 
+  static appendNavigationPermissions(req: Request, permission: string): void {
+    const session: SuppressionSession = this.getSuppressionSession(req)!;
+    const navigationPermissions: string[] | undefined = session.navigationPermissions;
+
+    if (!navigationPermissions) {
+      session.navigationPermissions = [permission]
+    } else if (!navigationPermissions.includes(permission)) {
+      session.navigationPermissions = session.navigationPermissions.concat([permission])
+    }
+
+    this.setSuppressionSession(req, session);
+  }
```

Comment:
This can be simplified, although i don't really see the need for a separate function since the `setSuppressionSession` could be used to update the navigation permissions (which is part of the suppression session after all)
```suggestion
  static appendNavigationPermissions(req: Request, permission: string): void {
    const session: SuppressionSession = this.getSuppressionSession(req)!;
    const navigationPermissions: string[] = session.navigationPermissions || [];
    session.navigationPermissions = [...navigationPermissions, permission]
    this.setSuppressionSession(req, session);
  }
```

Reply:
the reason its a function of its own is so that we could handle undefined navigation lists (which is the default value when the session is created)

Reply:
your solution doesn't seem to handle duplicates, something we avoided with 

```
else if (!navigationPermissions.includes(permission)) {
      session.navigationPermissions = session.navigationPermissions.concat([permission])
}
```

Reply:
I would also say that it is very nice to be able to just change the navigation list accross all controllers, rather than set the whole session object

Reply:
I think I will keep the function for the reason mentioned above, but I will do some refactoring inspired by your suggestions.

Reply:
looks good 👍